From a52b50bab7310b571dfffe37e9210fd7fe479d3b Mon Sep 17 00:00:00 2001
From: LuK1337 <priv.luk@gmail.com>
Date: Sun, 2 Jun 2019 20:15:49 +0200
Subject: [PATCH 11/15] Settings: appops: Drop GET_UNINSTALLED_PACKAGES from
 getApplicationInfo()

* Not sure why this flag was even there in the first place
  but removing it fixes an issue where app crashes due to
  ApplicationInfo.sourceDir being null.
* The easiest way to reproduce that issue is to remove
  /data/app/{} of any app listed in 'bootup' category
  then after a reboot Settings app will crash when trying
  to show said category.

Change-Id: I96e83a35b732b426a8e1a94604549172d2e9f03c
---
 src/com/android/settings/applications/appops/AppOpsState.java | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/com/android/settings/applications/appops/AppOpsState.java b/src/com/android/settings/applications/appops/AppOpsState.java
index 7684c4629e..68903a359b 100644
--- a/src/com/android/settings/applications/appops/AppOpsState.java
+++ b/src/com/android/settings/applications/appops/AppOpsState.java
@@ -607,8 +607,7 @@ public class AppOpsState {
         if (appInfo == null) {
             try {
                 appInfo = mPm.getApplicationInfo(packageName,
-                        PackageManager.GET_DISABLED_COMPONENTS
-                        | PackageManager.GET_UNINSTALLED_PACKAGES);
+                        PackageManager.GET_DISABLED_COMPONENTS);
             } catch (PackageManager.NameNotFoundException e) {
                 Log.w(TAG, "Unable to find info for package " + packageName);
                 return null;
-- 
2.17.1

